io/common/include/detray/io/file_handle.hpp

```diff
@@ -0,0 +1,66 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2023 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// System include(s)
+#include <cassert>
+#include <cstdint>
+#include <fstream>
+#include <ios>
+#include <iostream>
+#include <limits>
+#include <string>
+
+namespace detray::io::detail {
+
+/// Wrapper around a file stream
+class file_handle final {
+
+    public:
+    /// All writers must define a file name
+    file_handle() = delete;
+
+    /// File gets created with a @param name and @param extension
+    file_handle(const std::string &name = "",
+                const std::string &extension = "txt",
```

Comment:
:thinking: It could be a better idea to use [std::string_view](https://en.cppreference.com/w/cpp/string/basic_string_view) for these parameters.